src/basic/virt.c

```diff
@@ -521,29 +519,25 @@ int detect_container(void) {
                         return -EIO;
 
                 if (!startswith(t, "(1,")) {
-                        r = VIRTUALIZATION_CONTAINER_OTHER;
+                        v = VIRTUALIZATION_CONTAINER_OTHER;
                         goto finish;
                 }
         } else if (r != -ENOENT)
                 return r;
 
         /* If that didn't work, give up, assume no container manager. */
-        r = VIRTUALIZATION_NONE;
+        v = VIRTUALIZATION_NONE;
         goto finish;
 
 translate_name:
-        for (j = 0; j < ELEMENTSOF(value_table); j++)
-                if (streq(e, value_table[j].value)) {
-                        r = value_table[j].id;
-                        goto finish;
-                }
-
-        r = VIRTUALIZATION_CONTAINER_OTHER;
+        v = container_from_string(e);
+        if (v < 0)
+                v = VIRTUALIZATION_CONTAINER_OTHER;
 
 finish:
         log_debug("Found container virtualization %s.", virtualization_to_string(r));
```

Comment:
I think you meant `virtualization_to_string(v)` here (`v` instead of `r`)?

Reply:
Ouch.

Reply:
Thank you for finding that. I've force-pushed the revised commit.